online-shop/src/app/modules/shopping-cart/components/containers/shopping-cart-details/shopping-cart-details.component.ts

```diff
@@ -0,0 +1,31 @@
+import { Component } from '@angular/core';
+import { mockProducts } from 'src/app/mocks/products.mocks';
+import { Product } from 'src/app/modules/shared/types/products.types';
+
+@Component({
+  selector: 'app-shopping-cart-details',
+  templateUrl: './shopping-cart-details.component.html',
+  styleUrls: ['./shopping-cart-details.component.scss']
+})
+export class ShoppingCartDetailsComponent {
+  products: Product[] = mockProducts;
+  quantities: number[] = [];
+  productId:number | undefined;
+
+  ngOnInit() {
```

Comment:
If you're planning on using the on init lifecycle hook, your component must extend the OnInit interface. I suggest, however, using the OnChanges lifecycle hook for things that can potentially change.